esx_service/vmdk_ops.py

```diff
@@ -1075,22 +1084,109 @@ def load_vmci():
    else:
        lib = CDLL(os.path.join(LIB_LOC, "libvmci_srv.so"), use_errno=True)
 
+def getLock(lockname):
+    '''
+    Create or return a existing lock identified by lockname.
+    '''
+    global rsrcLocks
+    global managerLock
+
+    if not rsrcLocks:
```

Comment:
@andrewjstone 
You're right. 
Before moving the request composition to the worker thread (execRequestThread) i was explicitly initializing them through getLock() in the main thread before entering the loop.

I'll address those.

Thanks! 👍